packages/hardhat-vyper/src/compiler.ts

```diff
@@ -7,10 +8,21 @@ export class Compiler {
    *
    * @param inputPaths array of paths to contracts to be compiled
    */
-  public async compile(inputPaths: string[]) {
+  public async compile(inputPaths: string[], settings?: VyperSettings) {
```

Comment:
defaulting the `settings` parameter to `{}` would allow us to avoid the optional chaining operator (`?.`) after every usage of `settings` in the function.